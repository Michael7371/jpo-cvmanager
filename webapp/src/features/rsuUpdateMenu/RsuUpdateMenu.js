import React from 'react'
import { confirmAlert } from 'react-confirm-alert'
import RsuUpdateItem from '../../components/RsuUpdateItem'
import 'react-confirm-alert/src/react-confirm-alert.css'
import {
  selectOsUpdateAvailable,
  selectFwUpdateAvailable,

  // actions
  updateRsuData,
  performOSUpdate,
  performFWUpdate,
} from './rsuUpdateMenuSlice'
import { useSelector, useDispatch } from 'react-redux'

import './rsuUpdateMenu.css'

const RsuUpdateMenu = (props) => {
  const dispatch = useDispatch()
  const osUpdateAvailable = useSelector(selectOsUpdateAvailable)
  const fwUpdateAvailable = useSelector(selectFwUpdateAvailable)

  const verifyOS = {
    title: 'RSU OS Update',
    message: "Are you sure you want to update the RSU's operating system?",
    buttons: [
      {
        label: 'Yes',
        onClick: () => dispatch(performOSUpdate()),
      },
      {
        label: 'No',
        onClick: () => {},
      },
    ],
    childrenElement: () => <div />,
    closeOnEscape: true,
    closeOnClickOutside: true,
    keyCodeForClose: [8, 32],
    willUnmount: () => {},
    afterClose: () => {},
    onClickOutside: () => {},
    onKeypressEscape: () => {},
  }

  const verifyFW = {
    title: 'RSU Firmware Update',
    message: "Are you sure you want to update the RSU's firmware?",
    buttons: [
      {
        label: 'Yes',
        onClick: () => dispatch(performFWUpdate()),
      },
      {
        label: 'No',
        onClick: () => {},
      },
    ],
    childrenElement: () => <div />,
    closeOnEscape: true,
    closeOnClickOutside: true,
    keyCodeForClose: [8, 32],
    willUnmount: () => {},
    afterClose: () => {},
    onClickOutside: () => {},
    onKeypressEscape: () => {},
  }

  const handleUpdateOS = () => {
    confirmAlert(verifyOS)
  }

  const handleUpdateFW = () => {
    confirmAlert(verifyFW)
  }

  return (
    <div id="snmpdiv">
      <div id="updatediv">
        <h2 id="snmpheader">Administrator Update</h2>

        <button id="updatebtn" onClick={() => dispatch(updateRsuData(props.ipList))}>
          Check For Updates
        </button>

        {Object.entries(props.ipList).map((rsu) => (
          <RsuUpdateItem
            key={rsu[0]}
            ip={rsu[1]['properties']['Ipv4Address']}
            osUpdateAvailable={osUpdateAvailable}
            fwUpdateAvailable={fwUpdateAvailable}
            handleUpdateOS={handleUpdateOS}
            handleUpdateFW={handleUpdateFW}
          />
        ))}
      </div>
    </div>
  )
}

export default RsuUpdateMenu
