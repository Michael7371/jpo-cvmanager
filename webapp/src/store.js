import { configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddUserReducer from './features/adminAdduser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import menuReducer from './features/menu/menuSlice'
import rsuUpdateMenuReducer from './features/rsuUpdateMenu/rsuUpdateMenuSlice'

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminRsuTab: adminRsuTabReducer,
      adminUserTab: adminUserTabReducer,
      menu: menuReducer,
      rsuUpdateMenu: rsuUpdateMenuReducer,
    },
    preloadedState,
  })
}
