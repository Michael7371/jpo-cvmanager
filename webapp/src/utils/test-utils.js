/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

export { replaceChaoticIds }
