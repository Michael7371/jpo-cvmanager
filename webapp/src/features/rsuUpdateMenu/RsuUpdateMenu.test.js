import React from 'react'
import { render } from '@testing-library/react'
import RsuUpdateMenu from './RsuUpdateMenu'
import { Provider } from 'react-redux'
import { setupStore } from '../../store'
import { replaceChaoticIds } from '../../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(
    <Provider store={setupStore({})}>
      <RsuUpdateMenu ipList={[]} />
    </Provider>
  )

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
