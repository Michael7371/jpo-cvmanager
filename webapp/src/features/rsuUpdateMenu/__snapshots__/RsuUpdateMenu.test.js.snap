// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    id="snmpdiv"
  >
    <div
      id="updatediv"
    >
      <h2
        id="snmpheader"
      >
        Administrator Update
      </h2>
      <button
        id="updatebtn"
      >
        Check For Updates
      </button>
    </div>
  </div>
</div>
`;
